editor/src/components/canvas/canvas-strategies/strategies/absolute-resize-bounding-box-strategy.tsx

```diff
@@ -237,6 +254,232 @@ export function absoluteResizeBoundingBoxStrategy(
+    EP.isDescendantOf(element.elementPath, path),
+  )
+  for (const element of descendants) {
+    const constraintsArray = getSafeGroupChildConstraintsArray(allElementProps, element.elementPath)
+    const constraints = {
+      width: isDimensionConstrained(jsxMetadata, element.elementPath, constraintsArray, 'width'),
+      height: isDimensionConstrained(jsxMetadata, element.elementPath, constraintsArray, 'height'),
+      top: constraintsArray.includes('top'),
+      bottom: constraintsArray.includes('bottom'),
+      left: constraintsArray.includes('left'),
+      right: constraintsArray.includes('right'),
+    }
+
+    lockedWidth ||= constraints.width && constraints.left && constraints.right
+    lockedHeight ||= constraints.height && constraints.top && constraints.bottom
+
+    const isConstrained =
+      constraints.top ||
+      constraints.bottom ||
+      constraints.height ||
+      constraints.left ||
+      constraints.right ||
+      constraints.width
+
+    if (
+      isConstrained &&
+      element.localFrame != null &&
+      isFiniteRectangle(element.localFrame) &&
+      element.globalFrame != null &&
+      isFiniteRectangle(element.globalFrame)
+    ) {
+      const constrainedParentFrame = MetadataUtils.getFrameInCanvasCoords(
+        EP.parentPath(element.elementPath),
+        jsxMetadata,
+      )
+      const offsetDiffX =
+        constrainedParentFrame != null && isFiniteRectangle(constrainedParentFrame)
+          ? constrainedParentFrame.x - originalFrame.x
+          : 0
+      const offsetDiffY =
+        constrainedParentFrame != null && isFiniteRectangle(constrainedParentFrame)
+          ? constrainedParentFrame.y - originalFrame.y
+          : 0
+
+      constrainedSizes.push({
+        width: getBoundDimension(
+          constraints.left,
+          constraints.right,
+          constraints.width,
+          originalFrame.width,
+          element.localFrame.x,
+          element.localFrame.width + offsetDiffX,
+        ),
+        height: getBoundDimension(
+          constraints.top,
+          constraints.bottom,
+          constraints.height,
+          originalFrame.height,
+          element.localFrame.y,
+          element.localFrame.height + offsetDiffY,
+        ),
+      })
+    }
+  }
+
+  return {
+    constrainedSizes,
+    lockedWidth,
+    lockedHeight,
+  }
+}
+
+function getBoundDimension(
+  minBound: boolean,
+  maxBound: boolean,
+  dimensionBound: boolean,
+  originalDimension: number,
+  frameOffset: number,
+  frameDimension: number,
+): number {
+  if (minBound && maxBound) {
+    return originalDimension
+  } else if (minBound && dimensionBound) {
+    return frameOffset + frameDimension
+  } else if (minBound) {
+    return frameOffset
+  } else if (maxBound && dimensionBound) {
+    return originalDimension - frameOffset
+  } else if (maxBound) {
+    return originalDimension - frameDimension
+  } else if (dimensionBound) {
+    return frameDimension
+  } else {
+    return 0
```

Comment:
Shouldn't this return a `null` or some such? The zero looks like it might end up being the root of some weird bugs in the future.

Reply:
that's a good point, I made it nullable here [`584fe8a` (#4229)](https://github.com/concrete-utopia/utopia/pull/4229/commits/584fe8a3c01dc7b7cdc1f79ba0ec0c53be169e54)